include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -527,6 +539,48 @@ struct OrtEp {
   OrtStatus*(ORT_API_CALL* OnRunEnd)(_In_ OrtEp* this_ptr,
                                      _In_ const OrtRunOptions* run_options,
                                      _In_ bool sync_stream);
+
+  /** \brief Called by ORT to determine the compatibility of a compiled model with the EP.
+   *
+   * The application determines whether a given model compatibility (other than unsupported) should cause
+   * session-creation to fail.
+   *
+   * \param[in] this_ptr The OrtEp instance.
+   * \param[in] graph The top-level graph for the model.
+   * \param[out] model_compatibility Output parameter set to the OrtCompiledModelCompatibility enum value that
+   *                                 describes the compatibility of the model with the EP.
+   *
+   * \note Implementation of this function is optional. If not implemented, ORT assumes a result of
+   * OrtCompiledModelCompatibility_SUPPORT_UNKNOWN.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  OrtStatus*(ORT_API_CALL* GetCompiledModelCompatibility)(_In_ OrtEp* this_ptr,
+                                                          _In_ const OrtGraph* graph,
+                                                          _Out_ OrtCompiledModelCompatibility* model_compatibility);
+
+  /** \brief Generate a string with details about the EP stack used to produce a compiled model.
+   *
+   * This function generates a compatibility information string that contains details about the execution provider
+   * used to compile a given model. This string can later be used with ValidateCompiledModelCompatibilityInfo
+   * to determine if a compiled model is compatible with the EP.
+   * 
+   * The returned string should be a null-terminated, UTF-8 encoded string. ORT will copy it.
+   *
+   * \param[in] this_ptr The OrtEp instance.
+   * \param[in] graph The OrtGraph instance for which to generate compatibility information.
+   * \param[in] model_metadata The OrtModelMetadata instance containing model metadata.
+   * \return the compatibility information string produced by the EP
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  const char*(ORT_API_CALL* GenerateCompiledModelCompatibilityInfoString)(_In_ OrtEp* this_ptr,
+                                                                          _In_ const OrtGraph* graph,
+                                                                          _In_ const OrtModelMetadata* model_metadata);
```

Comment:
@skottmckay question here about the idea of persisting the compatibility information in the model metadata: would this be something added to the custom metadata map? Still learning about metadata in ORT- the alternative was adding it as a peer value to `producer_name`, et. al., which is quite a bit more surgery throughout the codebase. I'm moving in the direction of the former but wanted to cross-check.